portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java

```diff
@@ -33,12 +33,12 @@ public void init(){
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int numComments = getMax(request); 
+    int maxComments = Integer.parseInt(request.getParameter("max")); 
     Query query = new Query("Comment").addSort("timestamp_millis", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
     List<Comment> comments = new ArrayList<>();
-    if(numComments < 4){
-      for (Entity entity : results.asList(FetchOptions.Builder.withLimit(numComments))) {
+    if(maxComments != -1){
```

Comment:
This can be a little cleaner if you make the conditional only do the work specific to the branch. My recommendation:

```java
FetchOptions fetchOpts =  FetchOptions.Builder.withDefaults();
if (maxComments != -1) {
  fetchOpts = FetchOptions.Builder.withLimit(maxComments))
}
for (Entity entity : results.asList(fetchOpts)) {
  // ...
}
```